stdlib/asyncio/locks.pyi

```diff
@@ -19,13 +18,12 @@ if sys.version_info >= (3, 9):
 else:
     class _ContextManager:
         def __init__(self, lock: Lock | Semaphore) -> None: ...
-        def __enter__(self) -> object: ...
+        def __enter__(self) -> None: ...
         def __exit__(self, *args: Any) -> None: ...
 
     class _ContextManagerMixin:
-        def __init__(self, lock: Lock | Semaphore) -> None: ...
         # Apparently this exists to *prohibit* use as a context manager.
-        def __enter__(self) -> object: ...
+        def __enter__(self) -> NoReturn: ...
```

Comment:
this always raises a RuntimeError

Reply:
It might be better to comment out `__enter__` and `__exit__`. Type checkers will emit an error either way, but errors emitted from the `NoReturn` trick aren't usually that good. See also https://github.com/python/typing/issues/1043.